### ISSUE TITLE
EXC_BAD_ACCESS on Dynamic List on a new Object created during migration

### ISSUE THREAD

**Issue by Legoless** [realm/realm-cocoa#5789]
## Goals
Create a new object during migration and copy list of objects, from an old object.

## Expected Results
Receive list of `DynamicObject` instances that is operatable from the new object.

## Actual Results
Crashes with `EXC_BAD_ACCESS` on line 571 of `RLMAccessor.mm`:

`RLMListBase *list = object_getIvar(obj, prop.swiftIvar); // This is the line that crashes`

## Steps to Reproduce
1. Create two Realm classes of type `A` and `B`.
2. Class `A` should contain a list of `B` objects named `someList`
3. Rename class `A` to `ReplacedA` with same properties.
4. During migration attempt to recreate the relation of objects on list of `A` to `ReplacedA`.
5. `ReplacedA` objects should be created dynamically during migration.
6. Pull out list of objects on removed `A` class without object definition.

## Code Sample

Object classes:
```
class B : Object {
    @objc dynamic var someString : String? = nil
}

/*class A : Object {
    @objc dynamic var someOtherString : String? = nil

    let someList = List<B>()
}*/

class ReplacedA : Object {
    @objc dynamic var someOtherString : String? = nil

    let someList = List<B>()
}
```

Example migration code:
```
migration.enumerateObjects(ofType: "A") { oldObject, newObject in
    guard let oldObject = oldObject else {
        return
    }
    let renamedObject = migration.create("ReplacedA", value: [:])

    //print(renamedObject) // - This also causes EXC_BAD_ACCESS crash.

    for property in oldObject.objectSchema.properties {
        if let value = oldObject[property.name], renamedObject.objectSchema.properties.filter({ $0.name == property.name }).first != nil {
            if let value = value as? List<DynamicObject>, value.count > 0 {
                let newList = renamedObject.dynamicList(property.name) // This is where it crashes
                //let newList = renamedObject[property.name] as! List<DynamicObject>
                
                for currentValue in value {
                    newList.append(DynamicObject(value: currentValue)) // If second commented option above is used, then it throws an exception here that DynamicObject is not of the same type, even though the cast to List<DynamicObject> is successful.
                }
            }
        }
    }
}
```

## Version of Realm and Tooling

Realm framework version: 3.4.0 / 3.6.0

Realm Object Server version: /

Xcode version: 9.3.1

iOS/OSX version: 11.3.1

Dependency manager + version: CocoaPods 1.5.2

**Comment by Legoless** [comment]
To add to this issue, even this line after `create` will crash with `EXC_BAD_ACCESS` on same line:

`print(renamedObject)`

**Comment by tgoyne** [comment]
`object[property.name] as! List<MigrationObject>` is the correct way to access List properties within a migration.

Creating new objects within a migration must be done with `migration.create(className, value: value)`. Adding the object returned from that to a List of matching type should work correctly.

**Comment by Legoless** [comment]
Yes, but `MigrationObject` is defined as an alias of `DynamicObject`, so technically should not make a difference.

As for the `migration.create(className, value: value)`, it is correct, I just forgot to add it when extracting code. :)

**Comment by Legoless** [comment]
In addition to this, it seems that anytime `migration.create()` is called and returned object is printed, it will crash if object schema has a `List`.

**Comment by bmunkholm** [comment]
PR in progress for this.

**Comment by tgoyne** [comment]
Fixed in 3.9.0
